src/state_transition/types/execution_payload.zig

```diff
@@ -303,16 +279,42 @@ pub const ExecutionPayloadHeader = union(enum) {
             inline .deneb, .electra => |payload_header| payload_header.excess_blob_gas,
         };
     }
+
+    pub fn clone(self: *const ExecutionPayloadHeader, allocator: Allocator, out: *ExecutionPayloadHeader) !void {
+        switch (self.*) {
+            .bellatrix => |header| {
+                try types.bellatrix.ExecutionPayloadHeader.clone(allocator, header, @constCast(out.bellatrix));
+            },
+            .capella => |header| {
+                try types.capella.ExecutionPayloadHeader.clone(allocator, header, @constCast(out.capella));
+            },
+            .deneb => |header| {
+                try types.deneb.ExecutionPayloadHeader.clone(allocator, header, @constCast(out.deneb));
+            },
+            .electra => |header| {
+                try types.electra.ExecutionPayloadHeader.clone(allocator, header, @constCast(out.electra));
+            },
+        }
+    }
+
+    pub fn deinit(self: *const ExecutionPayloadHeader, allocator: Allocator) void {
+        switch (self.*) {
+            .bellatrix => |header| @constCast(header).extra_data.deinit(allocator),
```

Comment:
`deinit` the whole header so that when new fields added to it in future forks we don't have to be worried about